interview-questions.md

```diff
@@ -4,64 +4,78 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is used to store code to reuse;
 
-  Researched answer:
+  Researched answer: A JS function is a block of code designed to perform a particular task
 
 
 
 2. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: map is used to go through each element of an array and run a function on it.
+  filter is used to filter out criteria and possibly reduces the array
 
-  Researched answer:
+  Researched answer: map - the provided callback to map modifies the array element and save them into the new array upon completion that array get returned as the mapped array.
+
+  filter - filter does not update the existing array it will return a new filtered array everytime.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log() - is used to print out working code to check if your code works or to see whats happening
+  while return is used to output a value from a function.
 
-  Researched answer:
+  Researched answer: console.log() is a function used to printo information to the console.
+  return on the other hand is a call to pass some value back up to where the call was made.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what you place in your function to execute.
```

Comment:
sure! i would also add that it is _outside information_ that your function requires in order to run